ext/semian/tickets.c

```diff
@@ -27,68 +73,17 @@ update_ticket_count(update_ticket_count_t *tc)
 
   return Qnil;
 }
-void
-configure_tickets(int sem_id, int tickets, int should_initialize)
-{
-  if (should_initialize) {
-    initialize_tickets(sem_id, tickets);
-  }
-
-  if (tickets > 0) {
-    configure_static_tickets(sem_id, tickets);
-  }
-}
 
-static void
-initialize_tickets(int sem_id, int tickets)
+static int
+calculate_quota_tickets (int sem_id, double quota)
 {
-  unsigned short init_vals[SI_NUM_SEMAPHORES];
+  int tickets = 0;
 
-  init_vals[SI_SEM_TICKETS] = init_vals[SI_SEM_CONFIGURED_TICKETS] = tickets;
-  init_vals[SI_SEM_LOCK] = 1;
-
-  if (semctl(sem_id, 0, SETALL, init_vals) == -1) {
-    raise_semian_syscall_error("semctl()", errno);
+  // Ensure that a worker for this process is registered
```

Comment:
It's worth adding to the comment here that we from Ruby-land guarantee that a worker only goes through this code once.